plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeConfig.java

```diff
@@ -32,6 +33,7 @@
     private DataSize maxPageStorageSize = DataSize.of(16, MEGABYTE);
     private int exchangeSinkBufferPoolMinSize = 10;
     private int exchangeSinkBuffersPerPartition = 2;
+    private DataSize exchangeSinkFileTargetSize = DataSize.of(1, GIGABYTE);
```

Comment:
It feels like it might make things easier if the file size doesn't exceed the threshold. 

For example with `exchangeSourceConcurrentReaders` set to `4` the system should create tasks with at least 4 files assigned (one for each reader) to achieve desired parallelism.

That can be done by setting the `fault-tolerant-execution-target-task-input-size` to `4GB` and the `exchangeSinkFileTargetSize` to `1GB`.

However if the `FileSystemExchangeSink` is allowed to create slightly larger files the task may get assigned only `3` files, reducing the parallelism. While it is possible to configure `fault-tolerant-execution-target-task-input-size` to something arbitrary larger (e.g.: `4.1GB`) it might be less convenient.

I would recommend changing the property name to something like `exchangeSinkMaxFileSize` and adjusting the check in `FileSystemExchangeSink` to make sure the file size doesn't cross the threshold.